src/main/java/com/sarality/db/query/SimpleQueryBuilder.java

```diff
@@ -60,15 +72,44 @@ private String getWhereClause() {
   }
 
   private String[] getArguments() {
-    QueryClause  clause = new CompoundQueryClause(operator, clauseList);
+    QueryClause clause = new CompoundQueryClause(operator, clauseList);
     List<String> argumentList = clause.getSelectionArguments();
     if (argumentList == null || argumentList.isEmpty()) {
       return null;
     }
     return argumentList.toArray(new String[argumentList.size()]);
   }
 
+  private String getGroupByClause() {
+    StringBuilder builder = new StringBuilder();
+    int ctr = 0;
+    for (Column column : groupByColumnList) {
+      if (ctr > 0) {
+        builder.append(",");
+      }
+      builder.append(column.getName());
+      ctr++;
+    }
+
+    return builder.toString();
+  }
+
+  private String[] getSelectColumns() {
```

Comment:
the column list consists of anything in the groupby and all the aggregate measures. so I iterate through both lists and combine them to create the column list. moved the stringification to this method.